hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -164,6 +165,10 @@ def test_function(self, data):
         if data.status >= Status.VALID:
             for k, v in data.target_observations.items():
                 self.best_observed_targets[k] = max(self.best_observed_targets[k], v)
+                # We track the best score observed for any *failing* test separately,
+                # to help users diagnose degree of failure with threshold problems.
```

Comment:
I'm a little unclear as to why we need to restrict ourselves to the failing tests here. Either there's a threshold (in which case the maximum is always had for a failing test) or there isn't (in which case the target score is irrelevant to the test). Why not just show the full range?

Reply:
Excellent point - you're right, there's no reason to add the restriction here.

Reply:
Good point!  Now we just report the highest score for each label in statistics :smile: